planner/core/expression_rewriter.go

```diff
@@ -1654,6 +1654,8 @@ func (er *expressionRewriter) betweenToExpression(v *ast.BetweenExpr) {
 	}
 	l.SetCharsetAndCollation(dstCharset, dstCollation)
 	r.SetCharsetAndCollation(dstCharset, dstCollation)
+	l = expression.FoldConstant(l)
+	r = expression.FoldConstant(r)
```

Comment:
Maybe we can check whether `er.tryFoldCounter > 0` to decide `tryFold` or just `fold`